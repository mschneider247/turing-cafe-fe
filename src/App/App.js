import React, { Component } from 'react';
import './App.css';
import ReservationContainer from '../ReservationContainer/ReservationContainer';
import Form from '../Form/Form'

class App extends Component {
  constructor() {
    super();
    this.state = {
      reservations: [],
    }
  }

  componentDidMount () {
    fetch('http://localhost:3001/api/v1/reservations')
      .then(data => data.json())
      .then(data => this.setState({ reservations: data}))
  }

  addReservation = (reservation) => {
    let currentReservations = [...this.state.reservations]
    currentReservations.push(reservation)
    this.setState({ reservations: currentReservations})
    this.postReservation(reservation)
  }

  postReservation = (reservation) => {
    const options = {
      method: 'POST',
      body: JSON.stringify(reservation),
      headers: {
        'Content-Type': 'application/json'
      }
    };
    fetch('http://localhost:3001/api/v1/reservations', options)
  }

  removeReservation = (id) => {
    let currentReservations = [...this.state.reservations]
    currentReservations.forEach((reservation, index) => {
      if (reservation.id === id){
        currentReservations.splice(index, 1)
      }
    });
    this.setState({ reservations: currentReservations})
    this.deleteReservation(id);
  }

  deleteReservation = (id) => {
    const options = {
      method: 'DELETE'
    }
    fetch(`http://localhost:3001/api/v1/reservations/${id}`, options)
  }

  render() {
    return (
      <div className="App">
        <h1 className='app-title'>Turing Cafe Reservations</h1>
        <div className='resy-form'>
          <Form addReservation={this.addReservation}/>
        </div>
        <div className='resy-container'>
          <ReservationContainer reservations={this.state.reservations} removeReservation={this.removeReservation}/>
        </div>
      </div>
    )
  }
}

export default App;
