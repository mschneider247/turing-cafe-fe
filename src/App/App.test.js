import React from 'react';
import { shallow } from 'enzyme';
import App from './App';

describe('App', () => {

  it('Should match the snapshot', () => {
    const wrapper = shallow(<App />)
    expect(wrapper).toMatchSnapshot();
  })

  it('Should update state when new reservations recieved', () => {
    const wrapper = shallow(<App />);
    const mockReservation = {
      name: 'Jim',
      time: '7',
      date: '08/13',
      guests: 34
    }
    wrapper.instance().addReservation(mockReservation);
    expect(wrapper.state('reservations')).toEqual([mockReservation]);
  })
})

