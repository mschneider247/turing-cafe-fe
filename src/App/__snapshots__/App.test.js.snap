// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Should match the snapshot 1`] = `
<div
  className="App"
>
  <h1
    className="app-title"
  >
    Turing Cafe Reservations
  </h1>
  <div
    className="resy-form"
  >
    <Form
      addReservation={[Function]}
    />
  </div>
  <div
    className="resy-container"
  >
    <ReservationContainer
      reservations={Array []}
    />
  </div>
</div>
`;
